src/app/identity.service.ts

```diff
@@ -143,34 +144,49 @@ export class IdentityService {
   }
 
   private handleSign(data: any): void {
-    const { id, payload: { encryptedSeedHex, transactionHex } } = data;
+    const { id, payload: { encryptedSeedHex, transactionHex, messageHex } } = data;
 
-    // This will tell us whether we need full signing access or just ApproveLarge
-    // level of access.
-    const requiredAccessLevel = this.getRequiredAccessLevel(transactionHex);
+    const seedHex = this.cryptoService.decryptSeedHex(encryptedSeedHex, this.globalVars.hostname);
 
-    // In the case that approve() fails, it responds with a message indicating
-    // that approvalRequired = true, which the caller can then uses to trigger
-    // the /approve UI.
-    if (!this.approve(data, requiredAccessLevel)) {
-      return;
-    }
+    if (transactionHex) {
```

Comment:
```suggestion
    if (transactionHex && messageHex ) {} // Some error
    if (transactionHex && !messageHex) {
```
Here we check if there's no messageHex too in accordance with destructive priority